addon/internal-session.js

```diff
@@ -44,59 +43,80 @@ export default Ember.ObjectProxy.extend(Ember.Evented, {
   },
 
   restore() {
-    return new Ember.RSVP.Promise((resolve, reject) => {
-      let restoredContent   = this.store.restore();
-      let { authenticator } = (restoredContent.authenticated || {});
-      if (!!authenticator) {
-        delete restoredContent.authenticated.authenticator;
-        getOwner(this).lookup(authenticator).restore(restoredContent.authenticated).then((content) => {
+    return new RSVP.Promise((resolve, reject) => {
+      this._callStoreAsync('restore').then((restoredContent) => {
+        let { authenticator } = (restoredContent.authenticated || {});
+        if (!!authenticator) {
+          delete restoredContent.authenticated.authenticator;
+          getOwner(this).lookup(authenticator).restore(restoredContent.authenticated).then((content) => {
+            this.set('content', restoredContent);
+            this._setup(authenticator, content).then(resolve, reject);
+          }, () => {
+            Ember.Logger.debug(`The authenticator "${authenticator}" rejected to restore the session - invalidating…`);
+            this.set('content', restoredContent);
+            this._clear().then(resolve, reject);
```

Comment:
This should always reject actually as the session isn't actually successfully restored (not actually authenticated now) in this case.